presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java

```diff
@@ -139,18 +143,25 @@ public ExecutorService createCachingHiveMetastoreExecutor(HiveConnectorId hiveCl
     @Provides
     public HiveCluster createHiveCluster(Injector injector, HivePluginConfig config)
     {
-        URI uri = checkNotNull(config.getMetastoreUri(), "metastoreUri is null");
-        String scheme = uri.getScheme();
-        checkArgument(!isNullOrEmpty(scheme), "metastoreUri scheme is missing: %s", uri);
+        List<URI> uris = checkNotNull(config.getMetastoreUri(), "metastoreUri is null");
+        checkArgument(uris.size() > 0, "metastoreUri must specify at least one URI");
+        String scheme = uris.get(0).getScheme();
+        checkArgument(!isNullOrEmpty(scheme), "metastoreUri scheme is missing: %s", uris.get(0));
```

Comment:
We should just drop discovery support since no one uses it. Do that as a commit before this one, then this will be much simpler.